app/services/return-requirements/submit-additional-submission-options.service.js

```diff
@@ -0,0 +1,100 @@
+'use strict'
+
+/**
+ * Orchestrates validating the data for `/return-requirements/{sessionId}/additional-submission-options` page
+ * @module AdditionalSubmissionOptionsService
+ */
+
+const AdditionalSubmissionOptionsPresenter = require('../../presenters/return-requirements/additional-submission-options.presenter.js')
+const AdditionalSubmissionOptionsValidator = require('../../validators/return-requirements/additional-submission-options.validator.js')
+const SessionModel = require('../../models/session.model.js')
+
+/**
+ * Orchestrates validating the data for `/return-requirements/{sessionId}/additional-submission-options` page
+ *
+ * It first retrieves the session instance for the returns requirements journey in progress.
+ *
+ * The validation result is then combined with the output of the presenter to generate the page data needed by the view.
+ * If there was a validation error the controller will re-render the page so needs this information. If all is well the
+ * controller will redirect to the next page in the journey.
+ *
+ * @param {string} sessionId - The id of the current session
+ * @param {Object} payload - The submitted form data
+ * @param {Object} yar - The Hapi `request.yar` session manager passed on by the controller
+ *
+ * @returns {Promise<Object>} If no errors it returns an empty object else the page data for the note page including the
+ * validation error details
+ */
+async function go (sessionId, payload, yar) {
+  const session = await SessionModel.query().findById(sessionId)
+  // Ensure 'additional-submission-options' is an array and assign it to options
+  const options = payload['additional-submission-options'] = _ensureIsArray(payload['additional-submission-options'])
```

Comment:
I've reverted the function to how I set it up before. I only changed it to this, so that SonarCloud would allow it through the "20 lines" rule.

I looked at `_handleOneOptionSelected()`, and it manipulates `payload` outside of it's scope, but this also means it doesn't have proper closure.

This is why I am using `_ensureIsArray()` which does the same thing, but returns properly scoped. I also wrote the function so it could potentially be used as a generic helper function.